Modules/Module_Import/Import/xASL_imp_DetermineSubjectStructure.m

```diff
@@ -153,10 +148,19 @@
         warning('x.SUBJECTS is undefined, data loading is going to fail...');
     end
     
-
+	% If tokenVisitAliases is not provided, we assume visits are not renamed and create a dummy tokenVisitAliases list
+	if ~isfield(x.modules.import.imPar, 'tokenVisitAliases') || isempty(x.modules.import.imPar.tokenVisitAliases)
+		% Create a unique list of sessions
+		listSessions = unique(listSessions);
+
+		% Exclude empty session names
+		listSessions = listSessions(~ismember(listSessions,''));
+		
+		% Create a list of unique session names
+		if ~isempty(listSessions)
+			x.modules.import.imPar.tokenVisitAliases = listSessions(:);
```

Comment:
Why first (:) and second not?

BTW it seems you want

x.modules.import.imPar.tokenVisitAliases = [listSessions listSessions];

(unless you're afraid that somehow listSessions got transposed

Reply:
nono. this is correct and on purpose.

we want tokenVisitAliases to be a cell array nVisits x 2.
listSessions comes as 1 x nVisits.

So by using listSessions(:) we turn it into nVisits x 1 irrespective if listSessions is nVisit x 1 or 1xnVisits.

So x.modules.import.imPar.tokenVisitAliases = listSessions(:);
gives an array nVisits x 1 - always.

This command then gives 
x.modules.import.imPar.tokenVisitAliases(:,2) = listSessions;
the second column that is identical to the first column and we receive an array nVisits x 2 as we want.

When the matrix already exists and we add a column, it makes no difference if we do
x.modules.import.imPar.tokenVisitAliases(:,2) = listSessions;
or
x.modules.import.imPar.tokenVisitAliases(:,2) = listSessions(:)
or 
x.modules.import.imPar.tokenVisitAliases(:,2) = listSessions'
when listSessions is a vector.

Just put a breakpoint there to see for yourself...

Reply:
I know this, read my message, in my last sentence I said "unless you're afraid that somehow listSessions got transposed" which is exactly what you explain to me here, that in the case of a transposed listSessions, you would transpose back. I'm fine either way, your code is a bit longer but more secure.

If the reason for transposed listSessions is a bug, then your code would not give an error but we would get an error later on.

No clue what is better or worse. You can resolve this.

Reply:
Yes - upon creation, it's transposed. So it's a normal behavior that should be fixed. A normal transpose should be fine. But I'd rather use this secure thing to avoid future bugs when somebody changes this up in the code.